### PR TITLE
Introduction to C++ namespace

### LINKED ISSUE #1415: Introduction to C++ namespace
### Brief Summary: 
A name can represent only one entity in each scope. Thus, in the same scope, there cannot be two variables with the same name as this can generate compiler errors. We may declare two variables or member functions that have the same name in the same scope using namespaces. This causes several functions to have the same name and all functions can be accessed from anywhere in the program by referencing the name of the namespace. This article will go through namespaces and how they are used in C++.

### Key Takeaways: 
1. What is a namespace.
2. Features of namespace.
3. How a namespace works in c++.
4. Declaration of namespace.

### References: 
N/A

### PR DESCRIPTION
**Pull request by Dawe-7**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1415

### CODE REVIEW

**Comment by cyberShaw** [comment]
@Dawe-7 Amazing article! Code indentation was missing in the code snippets that you've given. Since the changes were minor, I fixed it myself. There is a lack of code comments in the snippets. Please add them and we can move on to final review.

File: articles/introduction-to-c++-namespace/index.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+`using namespace std;`
+
+which is used to give access to the namespace called standard that includes C++ I/O objects `cout` and `cin`.
+
+####  Example 1
+Program to illustrate working of namespace:
+
+```c++
+namespace no1 {
```
Comment (cyberShaw) [inline articles/introduction-to-c++-namespace/index.md:77]: The below code snippet needs helpful code comments

File: articles/introduction-to-c++-namespace/index.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+200
+no3::sample::show()
+```
+In the program above we have described the same function name in two different namespaces, `no1` and `no2`. We have also defined and accessed the class `sample` in another namespace `no3`.  As we have mentioned earlier to access the functions of the various namespaces the scope resolution operator`::` is used. As shown in the program, we can access the function `fun()` by using the operator as `no1::fun()`.
+
+#### Example 2
+Program to illustrate the working of namespace with the same name:
+
+```c++
+#include <iostream>
```
Comment (cyberShaw) [inline articles/introduction-to-c++-namespace/index.md:124]: Code comments here too, please :D

File: articles/introduction-to-c++-namespace/index.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+using namespace std;
+
+namespace no1 {
+  int sample = 10;
+  namespace no2 {
+    namespace no3 {
+      int sample1 = sample;
```
Comment (cyberShaw) [inline articles/introduction-to-c++-namespace/index.md:173]: Comments needed here too

**Review by cyberShaw — CHANGES_REQUESTED** [review]

**Comment by Dawe-7** [comment]
Hi @cyberShaw, I've added a few comments in the code snippets that the user can use to follow the program. I hope it's good to go, Kindly review.
Thanks.

**Comment by cyberShaw** [comment]
> Hi @cyberShaw, I've added a few comments in the code snippets that the user can use to follow the program. I hope it's good to go, Kindly review.
> Thanks.

I just saw the comments, they are really helpful! Thank you for making the changes. This article is ready for final review.

**Comment by Dawe-7** [comment]
> > Hi @cyberShaw, I've added a few comments in the code snippets that the user can use to follow the program. I hope it's good to go, Kindly review.
> > Thanks.
> 
> I just saw the comments, they are really helpful! Thank you for making the changes. This article is ready for final review.

Okay, thanks so much.

**Review by cyberShaw — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@Dawe-7 
Just made some changes for flow during final review. 

cc @cyberShaw  🚀  

Please do review to make sure no meaning was changed and to see the changes themselves

**Pull request merged**